presto-main/src/main/java/io/prestosql/execution/scheduler/SourcePartitionedScheduler.java

```diff
@@ -357,7 +365,20 @@ else if (pendingSplits.isEmpty()) {
             // The build side blocks due to a full output buffer.
             // In the meantime the probe side split cannot be consumed since
             // builder side hash table construction has not finished.
-            overallNewTasks.addAll(finalizeTaskCreationIfNecessary());
+            boolean shouldLockdownTasks = true;
+            if (anyBlockedOnPlacements && !groupedExecution) {
+                Supplier<Collection<TaskStatus>> childTaskStatus = childTaskStatusSupplier.get();
+                if (childTaskStatus != null) {
+                    boolean anyChildTaskOutputBufferOverutilized = childTaskStatus.get().stream()
+                            .anyMatch(task -> !task.getState().isDone() && task.isOutputBufferOverutilized());
```

Comment:
`task.getState().isDone()` seems redundant. Task with `FLUSHING` state won't be done, so it seems that only finished tasks are done, which don't have any buffer anymore.

Reply:
ping. Is `task.getState().isDone()` redundant?

Reply:
I think it is redundant (we also can get tasks which are in `RUNNING` state too along with `FLUSHING` ones).
I had put it initially to avoid any miscalculations due to incorrect flag setting in finished task - ideally that should never happen. is it ok to assume that for all finished tasks the output buffer should come to be under utilized?

Reply:
> is it ok to assume that for all finished tasks the output buffer should come to be under utilized?

I think so.